discord/client.py

```diff
@@ -442,6 +442,30 @@ def _reconnect_ws(self):
     # login state management
 
     @asyncio.coroutine
+    def login_with_token(self, token):
+        self.token = token
+        if self.token is None:
+            raise LoginFailure('No Token Specified.')
+            yield from self.close()
+        else:
+            self.headers['authorization'] = token
+            self._is_logged_in.set()
+            try:
+                self.loop.run_until_complete(self.connect())
```

Comment:
the original ``run`` / ``start`` does not work because they require a email and a password, The New type of account does not HAVE a email and a password just a token.